packages/protocol/contracts/tokenvault/ERC1155Vault.sol

```diff
@@ -218,34 +222,51 @@ contract ERC1155Vault is BaseNFTVault, ERC1155ReceiverUpgradeable {
             tokenIds: tokenIds,
             amounts: amounts
         });
+
+        return ERC1155Vault.onMessageRecalled.selector;
     }
 
-    function onERC1155Received(
+    function onERC1155BatchReceived(
         address,
         address,
-        uint256,
-        uint256,
+        uint256[] calldata,
+        uint256[] calldata,
         bytes calldata
     )
         external
         pure
         returns (bytes4)
     {
-        return IERC1155Receiver.onERC1155Received.selector;
+        return IERC1155Receiver.onERC1155BatchReceived.selector;
     }
 
-    function onERC1155BatchReceived(
+    function onERC1155Received(
         address,
         address,
-        uint256[] calldata,
-        uint256[] calldata,
+        uint256,
+        uint256,
         bytes calldata
     )
         external
         pure
         returns (bytes4)
     {
-        return IERC1155Receiver.onERC1155BatchReceived.selector;
+        return IERC1155Receiver.onERC1155Received.selector;
+    }
+
+    /**
+     * @dev See {IERC165-supportsInterface}.
+     */
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(ERC1155ReceiverUpgradeable)
+        returns (bool)
+    {
+        return interfaceId == type(ERC1155ReceiverUpgradeable).interfaceId
+            || interfaceId == ERC1155Vault.onMessageRecalled.selector
```

Comment:
why `type(ERC1155ReceiverUpgradeable).interfaceId` is so different from `ERC1155Vault.onMessageRecalled.selector`, aren't they both interfaces?